snuba/web/query.py

```diff
@@ -81,7 +82,7 @@ def _run_query_pipeline(
     - Providing the newly built Query, processors to be run for each DB query and a QueryRunner
       to the QueryExecutionStrategy to actually run the DB Query.
     """
-
+    assert isinstance(request.query, LogicalQuery)
```

Comment:
I think the right fix here is to execute line 86 to 113 is the query is simple and jump to 113 instead if the query is composite as the composite query does not support extensions. The pipeline builder is already wired up to deal with both types of queries.
Up to you if you want to do it here.